apis/apiextensions/v1/composition_types.go

```diff
@@ -205,6 +211,7 @@ const (
 	PatchTypeFromCompositeFieldPath PatchType = "FromCompositeFieldPath" // Default
 	PatchTypePatchSet               PatchType = "PatchSet"
 	PatchTypeToCompositeFieldPath   PatchType = "ToCompositeFieldPath"
+	PatchTypeFromFn                 PatchType = "FromFn"
```

Comment:
https://github.com/crossplane/crossplane/pull/2298#discussion_r628663555

Cross-linking the above thread, which could be relevant in this context too. Would there ever be a case in which we wanted to patch "FromFn" back to the XR, not to the composed resource? If so, would we want to do that from the context of a particular composed resource - would the "FromFn" function ever take properties of the composed resource as inputs?

Reply:
I am not sure, but I think at least in the current scenario with `FromFn` having just the `RandomString` utility, I don't think there would be a need to communicate back a random string from the Composed resource to the XR.